src/meta-srv/src/procedure/region_failover/activate_region.rs

```diff
@@ -0,0 +1,252 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::time::Duration;
+
+use api::v1::meta::mailbox_message::Payload;
+use api::v1::meta::MailboxMessage;
+use async_trait::async_trait;
+use common_meta::instruction::{Instruction, InstructionReply, SimpleReply};
+use common_meta::{DatanodeId, RegionIdent};
+use common_telemetry::debug;
+use serde::{Deserialize, Serialize};
+use snafu::{ensure, OptionExt, ResultExt};
+
+use super::update_metadata::UpdateRegionMetadata;
+use super::{RegionFailoverContext, State};
+use crate::error::{
+    DeserializeFromJsonSnafu, Error, Result, RetryLaterSnafu, SerializeToJsonSnafu,
+    UnexpectedInstructionReplySnafu,
+};
+use crate::service::mailbox::{Channel, MailboxReceiver};
+
+#[derive(Serialize, Deserialize, Debug)]
+pub(super) struct ActivateRegion {
+    candidate: DatanodeId,
+}
+
+impl ActivateRegion {
+    pub(super) fn new(candidate: DatanodeId) -> Self {
+        Self { candidate }
+    }
+
+    async fn send_open_region_message(
+        &self,
+        ctx: &RegionFailoverContext,
+        failed_region: &RegionIdent,
+        timeout: Duration,
+    ) -> Result<MailboxReceiver> {
+        let instruction = Instruction::OpenRegion(failed_region.clone());
+
+        let msg = MailboxMessage {
+            id: 0, // "id" will be set by the mailbox.
+            subject: "Activate Region".to_string(),
+            from: "Metasrv".to_string(),
```

Comment:
I think adding the metasrv ip and port into `from` will be more useful.

Reply:
I don't know if the metasrv has some `identifier` etc. to represent itself. We can use it as `from`. The identifier contains IP and port would be good.